### PR TITLE
feature(core): add bid history tab in auction detail

### LINKED ISSUE #1829: `auctions` - history
#### What would you like to be added:
Need to implement auction history on the app as a 2nd iteration. 
- [x] https://github.com/DeFiCh/whale/pull/649
- [ ] https://github.com/DeFiCh/whale/pull/698

Need more analysis on this and to check on the API side.

#### Feasibility of design vs API response
UI: https://www.figma.com/file/SDIpZciuxJlUf4xf8yheiY/PRDCT---Mobile-Light-Wallet?node-id=5942%3A89354
  - [x] Bid number, Token amount, USD amount, Time of Bid, Bidder ID, Transaction on Scan URL exists in response

#### Checklist
- [ ] Auction History - API Integration 
- [ ] Auction History - Bid History UI with translations
- [ ] Auction History - Format Bid Time (time ago) e.g 10 mins ago

### LINKED ISSUE #1911: Loans - Add Vault Sort order
#### What would you like to be added:
- Add vault sorting order to help with e2e testing and also UX.
Order would be:
Active (Red)
Active (Orange)
Active (Green)
In Liquidation
Ready
Halted

### PR DESCRIPTION
**Pull request by kyleleow**
#### What kind of PR is this?:

/kind feature

#### What this PR does / why we need it:
- To display bid history of an auction batch

#### Which issue(s) does this PR fixes?:

Fixes #1829
Fixes #1911

#### Additional comments?:
- Will implement e2e in a diff PR, tracked in #1898 

#### Developer Checklist:

- [x] Read your code changes at least once
- [x] Tested on iOS/Android device (e.g, No crashes, library supported etc.)
- [x] No console errors on web
- [x] Tested on Light mode and Dark mode*
- [x] Your UI implementation visually matched the rendered design*
- [x] Unit tests*
- [ ] Added e2e tests*
- [x] Added translations*

### CODE REVIEW

File: mobile-app/app/screens/AppNavigator/screens/Auctions/helpers/SecondstoHm.ts [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+}
+
+export function secondsToHmDisplay (d: number): string {
+    const { h, m } = secondsToHm(d)
+    const hDisplay = h > 0 ? `${translate('components/BatchCard', '{{h}}h', { h })} ` : ''
+    const mDisplay = m >= 0 ? translate('components/BatchCard', '{{m}}m', { m: h > 0 ? padStart(m.toString(), 2, '0') : m }) : ''
+    return `${hDisplay}${mDisplay}`
+  }
+
+export function secondsToTimeAgo (d: number): string {
```
Comment (kyleleow) [inline mobile-app/app/screens/AppNavigator/screens/Auctions/helpers/SecondstoHm.ts:18]: @pierregee can you help to check if we can refactor the logic here and with `secondsToHmDisplay()`?

**Review by pierregee — CHANGES_REQUESTED** [review]

File: mobile-app/app/screens/AppNavigator/screens/Auctions/components/BidHistory.tsx [hunk 2]
```diff
@@ -0,0 +1,197 @@
...
+        <ThemedText
+          numberOfLines={1}
+          ellipsizeMode='middle'
+          style={tailwind('w-4/12 text-xs font-medium')}
+        >
+          {props.bidderAddress}
+        </ThemedText>
+      </View>
+      <View style={tailwind('flex flex-row justify-between')}>
+        <ActiveUSDValue price={new BigNumber(props.bidAmount).multipliedBy(props.loanActivePrice)} />
```
[the remark this answers is not in the document]
Comment (kyleleow) [reply mobile-app/app/screens/AppNavigator/screens/Auctions/components/BidHistory.tsx]: updated

File: mobile-app/app/screens/AppNavigator/screens/Auctions/components/BidHistory.tsx [hunk 3]
```diff
@@ -0,0 +1,197 @@
...
+      vaultId: props.vaultId,
+      liquidationHeight: props.liquidationHeight,
+      batchIndex: props.batchIndex,
+      client: client,
+      size: 200
+    }))
+  }, [blockCount])
+
+  return (
+    <ThemedFlatList
```
[the remark this answers is not in the document]
Comment (kyleleow) [reply mobile-app/app/screens/AppNavigator/screens/Auctions/components/BidHistory.tsx:51]: added arbitrary padding for non-mobile

File: mobile-app/cypress/integration/functional/wallet/auctions/auctions.spec.ts [hunk 4]
```diff
@@ -13,8 +15,87 @@ function generateBlockUntilLiquidate (): void {
...
+        return
+      }
+
+      retries += 1
+      cy.wait(5000)
+      runIfAuctionsIsAvailable(callback)
+    })
+  }
+
+  // const recoveryWords: string[] = []
```
[the remark this answers is not in the document]
Comment (kyleleow) [reply mobile-app/cypress/integration/functional/wallet/auctions/auctions.spec.ts:67]: updated

**Review by fullstackninja864 — DISMISSED** [review]

File: mobile-app/app/screens/AppNavigator/screens/Auctions/components/BidHistory.tsx [hunk 5]
```diff
@@ -0,0 +1,205 @@
...
+    <ThemedView
+      light={tailwind({ 'bg-white border-gray-100': props.isLatestBid, 'bg-gray-100 border-gray-100': !props.isLatestBid })}
+      dark={tailwind({ 'bg-gray-800 border-gray-900': props.isLatestBid, 'bg-gray-900 border-gray-800': !props.isLatestBid })}
+      style={tailwind('border rounded px-4 py-3 mb-1')}
+      testID={`bid_${props.bidIndex.toString()}`}
+    >
+      <View style={tailwind('flex flex-row justify-between mb-2 items-center')}>
+        <ThemedView
+          style={tailwind('px-1 rounded-sm')}
+          light={tailwind('bg-blue-500 text-white')}
```
Comment (thedoublejay) [inline mobile-app/app/screens/AppNavigator/screens/Auctions/components/BidHistory.tsx:104]: Should this be only blue when it's the latest build? Based from the latest designs

image

**Review by thedoublejay — CHANGES_REQUESTED** [review]

**Comment by kyleleow** [comment]
Reverted the padding changes on `tabs` component, will track the cropped tab label in another [issue](https://github.com/DeFiCh/wallet/issues/2311)

**Review by thedoublejay — APPROVED** [review]

**Pull request merged**